lib/cosmos/interfaces/linc_interface.rb

```diff
@@ -234,40 +233,13 @@ def read
         if @handshake_packet.identify?(packet.buffer(false))
           handshake_packet = @handshake_packet.clone
           handshake_packet.buffer = packet.buffer
-          my_handshake = LincHandshake.new(handshake_packet, @target_names[0])
+          linc_handshake = LincHandshake.new(handshake_packet, @target_names[0])
 
+          # Check for a local handshake
           if handshake_packet.read('origin') == "LCL"
-            # Update the current value table for this command
-            command = System.commands.packet(my_handshake.identified_command.target_name, my_handshake.identified_command.packet_name)
-            command.received_time = my_handshake.identified_command.received_time
-            command.buffer = my_handshake.identified_command.buffer
-            command.received_count += 1
-
-            # Put a log of the command onto the server for the user to see
-            Logger.info("External Command: " + System.commands.format(my_handshake.identified_command, System.targets[@target_names[0]].ignored_parameters))
-
-            # Log the command to the command log(s)
-            @packet_log_writer_pairs.each do |packet_log_writer_pair|
-              packet_log_writer_pair.cmd_log_writer.write(my_handshake.identified_command)
-            end
+            handle_local_handshake(linc_handshake)
           else
-            # This is a remote packet (sent from here).
-            # Add to the array of handshake packet responses (only if handshakes are enabled).
-            # The mutex is required by the command task due to the way it
-            # first looks up the handshake before removing it.
-            if @handshake_enabled
-              @handshakes_mutex.synchronize do
-                @handshakes.push(my_handshake)
-                if @handshakes.length > MAX_CONCURRENT_HANDSHAKES
-                  len = @handshakes.length
-                  @handshakes = []
-                  raise "The handshakes response array has grown to #{len}. Clearing all handshakes!"
-                end
-                # Tell all waiting commands to take a look
-                @handshakes_resource.broadcast
-              end # @handshakes_mutex.synchronize
-            end # if @handshake_enabled
-
+            handle_remote_handshake(linc_handshake) if @handshake_enabled
           end # if handshake_packet.read('origin') == "LCL"
```

Comment:
I didn't understand this at first either so I made explicit local and remote handshake handling methods.